### PR TITLE
New features

### PR DESCRIPTION
**Pull request by fepitre**
Add new features:
- allow configuring default lvm thin pool
- select which templates to install
- allow selecting default template

https://github.com/QubesOS/qubes-issues/issues/5763

### CODE REVIEW

File: org_qubes_os_initial_setup/ks/qubes.py [hunk 1]
```diff
@@ -269,13 +291,37 @@ def configure_default_kernel(self):
         self.run_command([
             '/usr/bin/qubes-prefs', 'default-kernel', default_kernel])
 
+    def configure_default_pool(self):
+        self.set_stage("Setting up default pool")
+        if self.vg_tpool:
+            self.run_command(['/usr/bin/qvm-pool', '--add', 'default', 'lvm_thin',
+                              '-o', 'volume_group=%s,thin_pool=%s,revisions_to_keep=2' % self.vg_tpool])
```
Comment (marmarek) [inline org_qubes_os_initial_setup/ks/qubes.py]: What about actually setting it as the default pool?

File: org_qubes_os_initial_setup/ks/qubes.py [hunk 2]
```diff
@@ -104,26 +140,21 @@ def __init__(self, name):
...
 
+        self.custom_pool = False
+        self.vg_tpool = None
+
         self.skip = False
 
+        self.default_template = None
+        self.templates_to_install = \
+            ['fedora', 'debian', 'whonix-gw', 'whonix-ws']
+
```
Comment (marmarek) [inline org_qubes_os_initial_setup/ks/qubes.py:158]: All the settings here needs to be also properly serialized/deserialized in kickstart file. See `__str__` and `handle_line` methods.

**Review by marmarek — CHANGES_REQUESTED** [review]
(not complete review yet)
Have you tested it with non-lvm install? Like btrfs one, or plain static partitions.

File: org_qubes_os_initial_setup/ks/qubes.py (hunk 1 above)
Comment (fepitre) [reply org_qubes_os_initial_setup/ks/qubes.py]: Yes we can enforce this more but looking at the code of core-admin, it sets default pool on multiple search and a pool called 'default' is the first one. So maybe better adding the pool with name 'vm' and set it as default?

**Comment by fepitre** [comment]
> (not complete review yet)
> Have you tested it with non-lvm install? Like btrfs one, or plain static partitions.

Yes I tried manual BTFS install and it skips and makes unsensitive and unfilled the choices. Normally there is enough conditions for skipping in case of non-present lvm.

File: org_qubes_os_initial_setup/ks/qubes.py [hunk 3]
```diff
@@ -99,7 +99,9 @@ class QubesData(AddonData):
 
     bool_options = (
         'system_vms', 'default_vms', 'whonix_vms', 'whonix_default', 'usbvm',
-        'usbvm_with_netvm', 'skip')
+        'usbvm_with_netvm', 'skip', 'fedora_available', 'debian_available',
+        'whonix_available'
```
Comment (marmarek) [inline org_qubes_os_initial_setup/ks/qubes.py]: I don't think those three are part of the configuration - rather system state. On the other hand - selected lvm pool is.

**Review by marmarek — CHANGES_REQUESTED** [review]

**Review by marmarek — APPROVED** [review]

**Pull request merged**